src/hex/roller.ts

```diff
@@ -0,0 +1,90 @@
+import * as Mustache from 'mustache';
+import {countMatches} from '../arrays';
+import {combineAll} from '../lang';
+import {RandomNumberGenerator} from '../rng';
+import {combineRolls, Roll, rollDie, Roller} from '../roller';
+import base from '../template';
+import {DieRollView} from '../view';
+import {
+    BLUT_ROLL_TABLE,
+    BONUS_ROLL_TABLE,
+    Dice,
+    DicePool,
+    dieRollImages,
+    ELIXIR_ROLL_TABLE,
+    Faces,
+    FLUCH_ROLL_TABLE,
+    HEXXEN_ROLL_TABLE,
+    interpretResult,
+    MALUS_ROLL_TABLE,
+    parseRollValues,
+    RollValues,
+    rollValuesMonoid,
+    SEGNUNG_ROLL_TABLE,
+} from './dice';
+import {SimpleParser} from './parser';
+import tpl from './template';
+
+export class HEXRoller extends Roller<Dice, Faces, DicePool> {
+
+    constructor(private rng: RandomNumberGenerator, command: string) {
+        super(command, [new SimpleParser()], true, true);
+    }
+
+    public roll(pool: DicePool): Array<Roll<Dice, Faces>> {
+        if (pool.BONUS >= pool.MALUS) {
```

Comment:
Don't know the system very well, can you explain the mechanics?

Reply:
The basic Rules for dice are here: http://hexxen1733-regelwiki.de/index.php/regel_die_wuerfel.html (I assume you understand german. :)
For this special case: The is only one kind of dice for bonus and malus. The Rule for them are:
* add up all bonuses and cap them at 5 (if more)
* add up all maluses and cap them at 5 (if more)
* if after that you have more bonus than malus, then substract the malus count from the bonus count and roll the remaining number of janus dice which count as bonus (which means the get added to you successes).
* if you have more malus than bonus then substract the other way around an roll the remaining number of janus dice which in this case count as malus (which means the get substracted from you successes).